internal/monitors/ecs/ecs.go

```diff
@@ -0,0 +1,262 @@
+	"github.com/signalfx/signalfx-agent/internal/utils/filter"
+	log "github.com/sirupsen/logrus"
+)
+
+const monitorType = "ecs-metadata"
+
+// MONITOR(ecs-metadata): This monitor reads container stats from a
+// [ECS Task Metadata Endpoint version 2](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v2.html). 
+// 
+// This currently does not support CPU share/quota metrics.
+
+var logger = log.WithFields(log.Fields{"monitorType": monitorType})
+
+func init() {
+	monitors.Register(monitorType, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `acceptsEndpoints:"false"`
+
+	// The URL of the ECS task metadata. Default is http://169.254.170.2/metadata, which is hardcoded by AWS for version 2.
+	MetadataEndpoint string `yaml:"metadataEndpoint" default:"http://169.254.170.2/metadata"`
+	// The URL of the ECS container stats. Default is http://169.254.170.2/stats, which is hardcoded by AWS for version 2.
+	StatsEndpoint string `yaml:"statsEndpoint" default:"http://169.254.170.2/stats"`
+	// The maximum amount of time to wait for API requests
+	TimeoutSeconds int `yaml:"timeoutSeconds" default:"5"`
+	// A mapping of container label names to dimension names. The corresponding
+	// label values will become the dimension value for the mapped name.  E.g.
+	// `io.kubernetes.container.name: container_spec_name` would result in a
+	// dimension called `container_spec_name` that has the value of the
+	// `io.kubernetes.container.name` container label.
+	LabelsToDimensions map[string]string `yaml:"labelsToDimensions"`
+	// A list of filters of images to exclude.  Supports literals, globs, and
+	// regex.
+	ExcludedImages []string `yaml:"excludedImages"`
+}
+
+// Monitor for ECS Metadata
+type Monitor struct {
+	Output  types.Output
+	cancel  func()
+	client  *http.Client
+	conf *Config
+	ctx context.Context
+	timeout time.Duration
+	taskDimensions map[string]string
+	containers map[string]ecs.Container
+	// key : container docker id, tells if stats for the container should be ignored.
+	// Usually the container was filtered out by excludedImages
+	// or container metadata is not received.
+	shouldIgnore map[string]bool
+	imageFilter filter.StringFilter
+}
+
+// Configure the monitor and kick off volume metric syncing
+func (m *Monitor) Configure(conf *Config) error {
+	var err error
+	m.imageFilter, err = filter.NewBasicStringFilter(conf.ExcludedImages)
+	if err != nil {
+		return errors.Wrapf(err, "Could not load excluded image filter")
+	}
+
+	m.conf = conf
+	m.timeout = time.Duration(conf.TimeoutSeconds) * time.Second
+	m.client = &http.Client{
+		Timeout: m.timeout,
+	}
+	m.ctx, m.cancel = context.WithCancel(context.Background())
+
+	isRegistered := false
+
+	utils.RunOnInterval(m.ctx, func() {
+		if !isRegistered {
+			m.containers = map[string]ecs.Container{}
+			m.shouldIgnore = map[string]bool{}
+			m.fetchContainers()
```

Comment:
Make this return the set of containers and assign it to `m.containers` instead of implicitly sharing the variable through the struct instance.

Reply:
Same thing with the black list (`shouldIgnore`).